myhpom/forms/choose_network.py

```diff
@@ -15,22 +12,24 @@ def clean(self):
         custom_provider = cleaned_data.get('custom_provider')
         health_network = cleaned_data.get('health_network')
 
-        if ((not custom_provider) and (not health_network)) or (custom_provider and health_network):
-            raise forms.ValidationError(
-                'Please either choose a network or enter a custom network.'
-            )
+        if self.instance and self.instance.state and self.instance.state.healthnetwork_set.exists():
+            if (not custom_provider and not health_network) or (custom_provider and health_network):
+                raise forms.ValidationError(
+                    'Please either choose a network or enter a custom network.'
+                )
+        else:
+            if not custom_provider:
+                raise forms.ValidationError('Please enter a custom network.')
 
         return cleaned_data
 
     def clean_health_network(self):
-        data = self.cleaned_data['health_network']
-        if data:
+        health_network_id = self.cleaned_data['health_network']
+        if health_network_id:
```

Comment:
No need for the if block here - since None will raise DoesNotExist as well...

Reply:
😨 nevermind - I forgot that this needs to supply a dual purpose...this needs to return None sometimes. Disregard.